### PR TITLE
[Core] VTU Output

### PR DESCRIPTION
**Pull request by philbucher**
Together with a friend (@phmkopp) I am working on a vtu library that we can both use in our codes (https://github.com/phmkopp/vtu11). The principle is the same as pybind (e.g. header-only, c++11, ...)

vtu has several advantages over vtk:
- mesh can be written only once
- compression of results is available (also but not only with zlib)
- Results will be written into one file in MPI, using MPI-IO (which is an MPI 2 feature, lets see how it will be integrated), compared to one file / core for vtk

for me the use case is for FileIO in CoSimulation

Note that both the library itself and the integration into Kratos is WIP, for now I copy pasted many things from the vtk-output. Ultimately I will refactor this properly

### CODE REVIEW

**Comment by adityaghantasala** [comment]
Is the implementation already writing the results out ? No right ?

**Comment by philbucher** [comment]
> Is the implementation already writing the results out ? No right ?

you are right, it does not

File: kratos/input_output/vtu_output.cpp [hunk 1]
```diff
@@ -0,0 +1,259 @@
+//    |  /           |
+//    ' /   __| _` | __|  _ \   __|
+//    . \  |   (   | |   (   |\__ `
+//   _|\_\_|  \__,_|\__|\___/ ____/
+//                   Multi-Physics
+//
+//  License:		 BSD License
+//					 Kratos default license: kratos/license.txt
+//
+//  Main authors:    Philipp Bucher (based on vtk_output.h)
```
Comment (loumalouomega) [inline kratos/input_output/vtu_output.cpp:10]: It is quite similar, could be done with a template?

**Review by loumalouomega — COMMENTED** [review]
One comment

File: kratos/input_output/vtu_output.cpp (hunk 1 above)
Comment (adityaghantasala) [reply kratos/input_output/vtu_output.cpp:10]: or may be a base class .. something like ParaviewOutputBaseClass

Comment (philbucher) [reply kratos/input_output/vtu_output.cpp:10]: I will probably refactor common functionalities into free functions. Even now there are many fct that don't require any class members

**Comment by philbucher** [comment]
This is far from being ready, but initial tests show a reduction in size with compression ~30% compared to vtk for a regular fluid case

**Comment by philbucher** [comment]
Closing as this is super outdated now

I implemented this nicely in https://github.com/philbucher/Kratos/tree/vtu-output but I don't have the time to push it atm
Hopefully after i submitted :)

**Comment by loumalouomega** [comment]
> Closing as this is super outdated now
> 
> I implemented this nicely in [philbucher/Kratos@`vtu-output`](https://github.com/philbucher/Kratos/tree/vtu-output?rgh-link-date=2021-11-04T12%3A00%3A46Z) but I don't have the time to push it atm Hopefully after i submitted :)

What is missing?

**Comment by philbucher** [comment]
> What is missing?

- Features (now I only support nodal historical variables)
- Tests
- Probably refactor the VTKOutput so that the VTUOutput and the VTKOutput can have the same baseclass

I am still working on this / improving it / testing it, will get back as soon as I have more time again

**Pull request closed without merging**